inf-terraform-aws/Jenkinsfile.template

```diff
@@ -0,0 +1,322 @@
+/* generated jenkins file used for building and deploying AWS-infrastructure in projects */
+
+def final projectId = '@project_id@'
+def final componentId = '@component_id@'
+def final credentialsId = "${projectId}-cd-cd-user-with-password"
+
+def sharedLibraryRepository
+def dockerRegistry
+
+// QS Variables
+def preFix
+def stackConfig
+def stackdestroy = false
+def stackdeploy  = false
+
+def awsAccessKeyId
+def awsSecretAccessKey
+def awsRegion
+
+def consulHttpAddr  
+def consulHttpToken 
+
+def awsAccessKeyIdTesting     
+def awsSecretAccessKeyTesting 
+def awsRegionTesting          
+
+def git_login_url
+def git_login_port
+def tfvarsjsonfile
+def tfvarsjson
+def tfconsulpath
+
+@Library('ods-jenkins-shared-library@master') _
+
+odsComponentPipeline(
+  imageStreamTag: 'ods/jenkins-agent-terraform:latest',
+  branchToEnvironmentMapping: [
+    'master': 'dev',
+    // 'release/': 'test'
+  ],
+  podVolumes: [
+    persistentVolumeClaim(mountPath: "/tmp/.terraform-state", claimName: "terraform-state", readOnly: false)
+  ]
+) { context ->
+
+    stageGetConfig(context)
+    stageInit(context)
+    stageDestroy(context)
+    stageTest(context)
+    stagePlan(context)
+    stageDeploy(context)
+    stageSmoke(context)
+    stageReport(context)
+
+  // odsComponentStageImportOpenShiftImageOrElse(context) {
+    /*
+    * if you want to introduce scanning, uncomment the below line
+    * and change the type in metadata.yml to 'ods'
+    *
+    * odsComponentStageScanWithSonar(context)
+    */
+  // odsComponentStageBuildOpenShiftImage(context)
+  //}
+  // odsComponentStageRolloutOpenShiftDeployment(context)
+}
+
+def getEnvSettings(def consulPath, def consulHttpAddr, def workspace, def region ) {
+  def config = [
+    "NO_PROXY=127.0.0.1,10.0.0.0/8,10.183.175.157,localhost,boehringer.com,eu.boehringer.com,am.boehringer.com",
+    "GIT_LOGIN_URL=git@bitbucket.biscrum.com",
+    "GIT_LOGIN_PORT=7999"
+  ]

```

Comment:
Hardcoded stuff should be removed. `NO_PROXY` should be baked into the image using cluster-wide config (this is done automatically).